src/libNode/PoWProcessing.cpp

```diff
@@ -91,63 +89,131 @@ bool Node::StartPoW(const uint64_t& block_num, uint8_t difficulty,
+    // Message = [8-byte block number] [1 byte difficulty level] [4-byte listening port] [33-byte public key]
+    // [8-byte nonce] [32-byte resulting hash] [32-byte mixhash] [64-byte Signature]
+    vector<unsigned char> powmessage
+        = {MessageType::DIRECTORY, DSInstructionType::POWSUBMISSION};
+    unsigned int cur_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint64_t>(powmessage, cur_offset, block_num,
+                                      sizeof(uint64_t));
+    cur_offset += sizeof(uint64_t);
 
-        m_mediator.m_selfKey.second.Serialize(powmessage, cur_offset);
-        cur_offset += PUB_KEY_SIZE;
+    Serializable::SetNumber<uint8_t>(powmessage, cur_offset, difficultyLevel,
+                                     sizeof(uint8_t));
+    cur_offset += sizeof(uint8_t);
 
-        Serializable::SetNumber<uint64_t>(powmessage, cur_offset,
-                                          winning_result.winning_nonce,
-                                          sizeof(uint64_t));
-        cur_offset += sizeof(uint64_t);
-
-        powmessage.insert(powmessage.end(), result_vec.begin(),
-                          result_vec.end());
-        cur_offset += BLOCK_HASH_SIZE;
-        powmessage.insert(powmessage.end(), mixhash_vec.begin(),
-                          mixhash_vec.end());
-        cur_offset += BLOCK_HASH_SIZE;
-
-        Signature sign;
-        if (!Schnorr::GetInstance().Sign(powmessage, m_mediator.m_selfKey.first,
-                                         m_mediator.m_selfKey.second, sign))
-        {
-            LOG_GENERAL(WARNING, "Failed to sign PoW");
-        }
-        sign.Serialize(powmessage, cur_offset);
+    Serializable::SetNumber<uint32_t>(powmessage, cur_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    cur_offset += sizeof(uint32_t);
 
-        deque<Peer> peerList;
+    m_mediator.m_selfKey.second.Serialize(powmessage, cur_offset);
+    cur_offset += PUB_KEY_SIZE;
 
-        for (auto const& i : *m_mediator.m_DSCommittee)
-        {
-            peerList.push_back(i.second);
-        }
+    Serializable::SetNumber<uint64_t>(powmessage, cur_offset, winningNonce,
+                                      sizeof(uint64_t));
+    cur_offset += sizeof(uint64_t);
+
+    powmessage.insert(powmessage.end(), powResultHash.begin(),
+                      powResultHash.end());
+    cur_offset += BLOCK_HASH_SIZE;
+    powmessage.insert(powmessage.end(), powMixhash.begin(), powMixhash.end());
+    cur_offset += BLOCK_HASH_SIZE;
 
-        P2PComm::GetInstance().SendMessage(peerList, powmessage);
+    Signature sign;
+    if (!Schnorr::GetInstance().Sign(powmessage, m_mediator.m_selfKey.first,
+                                     m_mediator.m_selfKey.second, sign))
+    {
+        LOG_GENERAL(WARNING, "Failed to sign PoW");
     }
+    sign.Serialize(powmessage, cur_offset);
 
+    deque<Peer> peerList;
```

Comment:
If only need to push_back, why don't use vector instead of deque, vector if more efficient if only need to push_back.

Reply:
alright. will do.